### ISSUE TITLE
Internal Error Logging In

### ISSUE THREAD

**Issue by Kyrluckechuck** [owncloud/core#24749]
![firefox_2016-05-20_12-01-04](https://cloud.githubusercontent.com/assets/7606153/15434314/3d1f811a-1e84-11e6-84cf-b71cb22f17df.png)
 
### Steps to reproduce
1. Go to cloud.website.com
2. Attempt to Login

### Expected behaviour
Tell us what should happen
- Should login without error
### Actual behaviour
Tell us what happens instead
- First login error "Internal Error Occurred"
- Attempt to login again will work
### Server configuration

**Operating system**:
 - Ubuntu 14.04.4 LTS

**Web server:**
 - Apache-2.4.7 (Sentora)

**Database:**
 - MySQL-14.14 (Sentora)

**PHP version:**
- PHP-5.5.9 (Sentora)

**ownCloud version:** (see ownCloud admin page)
- 9.0.0.19

**Updated from an older ownCloud or fresh install:**
- Fresh, files copied from 8.0.1

**Where did you install ownCloud from:**
- Web Installer, though all other dependencies installed now

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - comments: 0.2
  - dav: 0.1.5
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - music: 0.3.11
  - notifications: 0.2.3
  - ownbackup: 0.3.8
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - encryption
  - external
  - files_antivirus
  - files_external
  - user_external
  - user_ldap
```

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "oc0wy480z28l",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.website.com"
        ],
        "datadirectory": "\/var\/sentora\/hostdata\/kyrluckechuck\/public_html\/cloud_website_com\/data",
        "overwrite.cli.url": "http:\/\/cloud.website.com",
        "dbtype": "mysql",
        "version": "9.0.0.19",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_from_address": "name",
        "mail_domain": "website.com",
        "mail_smtphost": "smtp.gmail.com",
        "mail_smtpport": "587",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "tls",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "dbdriveroptions": {
            "1002": "SET wait_timeout = 28800"
        },
        "loglevel": 1,
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
No
**Are you using encryption:** yes/no
No
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

### Client configuration
**Browser:**
Firefox, Android Firefox, IE
**Operating system:**
Windows 7, Android 6.0.1 - BB Priv
### Logs
Nothing other than startup, no errors
#### Web server error log

Insert your webserver log here
```
Nothing
```
Insert your ownCloud log here
```
The file is over 1200 lines, please specify what to paste. Also contains sensitive data it seems
```
#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

Extra:
Not sure if it's relevant, but I'm unable to update to 9.0.1/9.0.2 as it says I'm on the latest Stable/Beta builds (doesn't matter which branch)/

**Comment by RealRancor** [comment]
@Kyrluckechuck 

> The file is over 1200 lines, please specify what to paste. Also contains sensitive data it seems

Please delete the complete content of the owncloud.log and then login once to only get the relevant logentries. "Please specify what to paste" makes no sense as a internal server error is just a generic errors which might be caused by tons of reasons.

> Not sure if it's relevant, but I'm unable to update to 9.0.1/9.0.2 as it says I'm on the latest Stable/Beta builds (doesn't matter which branch)

You should definitely update to 9.0.2. Just don't use the updater app (which is currently not shipping 9.0.2) and extract the 9.0.2 archive from https://owncloud.org/changelog/ to your webspace.

**Comment by Kyrluckechuck** [comment]
Alright, by deleting it and attempting to login, I'm presented with this:
```
{"reqId":"x2VG8ymRzM5bBdxr5Lc9","remoteAddr":"99.225.44.66","app":"handleLogin","message":"Exception: {\"Exception\":\"OCP\\\\PreConditionNotMetException\",\"Message\":\"\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/sentora\\\/hostdata\\\/kyrluckechuck\\\/public_html\\\/cloud_website_com\\\/lib\\\/private\\\/allconfig.php(222): OC\\\\DB\\\\Connection->setValues('preferences', Array, Array, Array)\\n#1 \\\/var\\\/sentora\\\/hostdata\\\/kyrluckechuck\\\/public_html\\\/cloud_website_com\\\/lib\\\/base.php(1079): OC\\\\AllConfig->setUserValue('Kyrluckechuck', 'core', 'timezone', 'America\\\/New_Yor...')\\n#2 \\\/var\\\/sentora\\\/hostdata\\\/kyrluckechuck\\\/public_html\\\/cloud_website_com\\\/lib\\\/base.php(969): OC::tryFormLogin()\\n#3 \\\/var\\\/sentora\\\/hostdata\\\/kyrluckechuck\\\/public_html\\\/cloud_website_com\\\/lib\\\/base.php(931): OC::handleLogin()\\n#4 \\\/var\\\/sentora\\\/hostdata\\\/kyrluckechuck\\\/public_html\\\/cloud_website_com\\\/index.php(39): OC::handleRequest()\\n#5 {main}\",\"File\":\"\\\/var\\\/sentora\\\/hostdata\\\/kyrluckechuck\\\/public_html\\\/cloud_website_com\\\/lib\\\/private\\\/db\\\/connection.php\",\"Line\":297}","level":3,"time":"2016-05-20T19:19:30+00:00"}
```

I'll update that soon, though that shouldn't be relevant, correct?

**Comment by Xenopathic** [comment]
Sounds like this was fixed with 9.0.1 in #23309. Please upgrade.

**Comment by Kyrluckechuck** [comment]
Hey there! Just wanted to confirm this was for sure fixed in 9.0.2 updating via the files as @RealRancor said. Thanks for closing @Xenopathic.
Also this fixed issues with syncing via Official Android OwnCloud app
